src/main/java/appeng/hooks/ticking/TickHandler.java

```diff
@@ -381,19 +376,19 @@ private void readyTiles(IWorld world) {
             // Chunks which are considered a border chunk will not "exist", but are loaded. Once this state changes they
             // will be readied.
             if (world.chunkExists(pos.x, pos.z) && chunkProvider.canTick(testBlockPos)) {
-                Queue<AEBaseTileEntity> queue = entry.getValue();
-
-                while (!queue.isEmpty()) {
-                    final AEBaseTileEntity bt = queue.poll();
+                Queue<AEBaseTileEntity> chunkQueue = worldQueue.remove(chunkPos);
+                if (chunkQueue == null) {
+                    continue; // This should never happen, chunk unloaded under our noses
+                }
 
+                for (AEBaseTileEntity bt : chunkQueue) {
```

Comment:
How does this ever remove the processed `AEBaseTileEntity` and not just readies them the next tick again? As far as I am aware, the `Iterator` for an `ArrayDeque` does not actually do a `poll`.

Also if it would, we should not rely on some special behaviour of a specific collection implementation. Otherwise someone will just look at it again, see we have a `Queue`, but never use it as queue and then just refactors it to a simple list for performance.